classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java

```diff
@@ -416,17 +472,31 @@ public TCollection<V> values() {
         @SuppressWarnings("unchecked")
         @SafeVarargs
         NEtriesMap(Entry<K, V>... data) {
-            Entry<K, V>[] table = new Entry[data.length];
+            this.data = toEntryArray(data);
+        }
+
+        @SuppressWarnings("unchecked")
+        NEtriesMap(TMap<? extends K, ? extends V> map) {
+            this.data = toEntryArray(map.entrySet().toArray(new TMap.Entry[0]));
+        }
+
+        /**
+         * Creates an array where the {@code Entry} elements are positioned in such a way they
+         * are compatible with the contract of {@link java.util.Map}.
+         */
+        @SuppressWarnings("unchecked")
+        private Entry<K, V>[] toEntryArray(Entry<K, V>[] entries) {
+            Entry<K, V>[] table = new Entry[entries.length];
```

Comment:
Does this line pass checkstyle?

Reply:
It does, I wouldn't dare to submit code which doesn't pass tests or checkstyle ;-). Is there something you want me to modify here? Maybe I'm missing the obvious but I don't think I see any style problems with it.

Reply:
I think the right hand side (`new Entry[]`) is seen as a unchecked cast but not as raw usage of parameterized class. The left hand side (`Entry<K, V>`) would be seen as raw usage of parameterized class if it was without type definition (so `Entry[] table` instead of `Entry<K, V>[] table`). In that case it would indeed likely trigger a checkstyle raw usage error. 

I thought about it a bit but I also don't really see an alternative way of doing it (with better generics) compared to the current implementation.